src/tracing/esp32_diagnostic_trace/BUILD.gn

```diff
@@ -20,6 +20,13 @@ config("tracing") {
   include_dirs = [ "include" ]
 }
 
+config("diagnostic_config") {
+  defines = [
+    "CONFIG_ENABLE_TRACES=1",
+    "CONFIG_ENABLE_METRICS=1",
```

Comment:
It's a bool type option, so it will either be defined or not. We are treating it as int here.